ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/JmsXAResourceRecoveryHelper.java

```diff
@@ -177,4 +168,10 @@ private void disconnect() throws XAException {
         }
     }
 
+    private void assertIfRecoveryScanWasStarted() throws XAException {
+        if (delegate == null) {
+            throw new XAException("Recovery scan has to be started");
```

Comment:
Also "Recovery scan has to be started" is not meaningful and it should be localizable (ie appear in our JTA log messages file).